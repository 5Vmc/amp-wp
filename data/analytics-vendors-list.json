{
    "acquialift": "Acquia Lift",
    "adobeanalytics": "Adobe Analytics",
    "adobeanalytics_nativeConfig": "Adobe Analytics (native config)",
    "afsanalytics": "AFS Analytics",
    "alexametrics": "Alexa Internet",
    "amplitude": "Amplitude",
    "appsflyer": "AppsFlyer",
    "atinternet": "AT Internet",
    "baiduanalytics": "Baidu Analytics",
    "blackcrowai": "Black Crow AI",
    "blueconic": "BlueConic",
    "bluetriangle": "Blue Triangle",
    "browsi": "Browsi",
    "burt": "Burt",
    "byside": "BySide",
    "captainmetrics": "Captain Metrics",
    "chartbeat": "Chartbeat",
    "clicky": "Clicky Web Analytics",
    "colanalytics": "colanalytics",
    "comscore": "comScore",
    "cxense": "Cxense",
    "deepbi": "Deep.BI",
    "dynatrace": "Dynatrace",
    "epica": "EPICA",
    "euleriananalytics": "Eulerian Analytics",
    "facebookpixel": "Facebook Pixel",
    "gemius": "Gemius",
    "gfksensic": "GfK Sensic",
    "googleadwords": "Google Ads",
    "googleanalytics": "Google Analytics",
    "gtag": "gtag",
    "": "Google Tag Manager",
    "ibeatanalytics": "Ibeat Analytics",
    "infonline": "INFOnline / IVW",
    "infonline_anonymous": "INFOnline anonymous",
    "infonline_base": "INFOnline base",
    "iplabel": "ip-label",
    "keen": "Keen",
    "kenshoo": "Kenshoo",
    "krux": "Krux",
    "linkpulse": "Linkpulse",
    "lotame": "Lotame",
    "mapp_intelligence": "Mapp Intelligence",
    "marinsoftware": "Marin Software",
    "mediametrie": "Médiamétrie",
    "mediarithmics": "mediarithmics",
    "mediator": "mediator",
    "memo": "Memo",
    "metrika": "Yandex Metrica",
    "moat": "Moat Analytics",
    "mobify": "Mobify",
    "moengage": "MoEngage",
    "mparticle": "mParticle",
    "mpulse": "SOASTA mPulse",
    "navegg": "Navegg",
    "newrelic": "New Relic",
    "nielsen": "Nielsen",
    "nielsen-marketing-cloud": "Nielsen Marketing Cloud",
    "oewa": "OEWA",
    "oewadirect": "oewadirect",
    "oracleInfinityAnalytics": "Oracle Infinity Analytics",
    "parsely": "Parsely",
    "permutive": "Permutive",
    "permutive-ampscript": "Permutive-ampscript",
    "piano": "Piano",
    "pinpoll": "Pinpoll",
    "piStats": "Pistats",
    "ppasanalytics": "Piwik PRO Analytics Suite",
    "pressboard": "Pressboard",
    "quantcast": "Quantcast Measurement",
    "rakam": "Rakam",
    "reppublika": "reppublika",
    "retargetly": "Retargetly",
    "rudderstack": "RudderStack",
    "segment": "Segment",
    "sensorsanalytics": "SensorsData",
    "shinystat": "ShinyStat",
    "snowplow": "Snowplow Analytics",
    "snowplow_v2": "Snowplow Analytics (v2)",
    "subscriptions-propensity": "subscriptions-propensity",
    "taboola": "Taboola",
    "tail": "Tail",
    "teaanalytics": "TEA Analytics",
    "tealiumcollect": "Tealium Collect",
    "top100": "Rambler/TOP-100",
    "topmailru": "Top.Mail.Ru",
    "treasuredata": "Treasure Data",
    "umenganalytics": "Umeng+ Analytics",
    "upscore": "Upscore",
    "vponanalytics": "Vpon Analytics",
    "webengage": "Webengage",
    "webtrekk_v2": "Webtrekk"
}