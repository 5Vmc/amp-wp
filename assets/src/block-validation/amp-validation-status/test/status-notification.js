/**
 * External dependencies
 */
import { act } from 'react-dom/test-utils';

/**
 * WordPress dependencies
 */
import { render, unmountComponentAtNode } from '@wordpress/element';
import { useSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import AMPValidationStatusNotification from '../status-notification';

jest.mock( '@wordpress/data/build/components/use-select', () => jest.fn() );

describe( 'AMPValidationStatusNotification', () => {
	let container;

	function setupUseSelect( overrides ) {
		useSelect.mockImplementation( () => ( {
			ampCompatibilityBroken: false,
			hasValidationErrors: false,
			isEditedPostNew: false,
			isFetchingErrors: false,
			...overrides,
		} ) );
	}

	beforeEach( () => {
		container = document.createElement( 'div' );
		document.body.appendChild( container );
	} );

	afterEach( () => {
		unmountComponentAtNode( container );
		container.remove();
		container = null;
	} );

	it( 'renders message when there are no AMP validation errors', () => {
		setupUseSelect();

		act( () => {
			render( <AMPValidationStatusNotification />, container );
		} );

		expect( container.innerHTML ).toMatchSnapshot();
		expect( container.innerHTML ).toContain( 'All issues reviewed or removed' );
		expect( container.querySelector( '.is-error' ) ).toBeNull();
	} );

	it( 'does not render when errors are being fetched', () => {
		setupUseSelect( {
			isFetchingErrors: true,
		} );

		act( () => {
			render( <AMPValidationStatusNotification />, container );
		} );

		expect( container.children ).toHaveLength( 0 );
	} );

	it( 'renders error message when AMP compatibility is broken', () => {
		setupUseSelect( {
			ampCompatibilityBroken: true,
		} );

		act( () => {
			render( <AMPValidationStatusNotification />, container );
		} );

		expect( container.innerHTML ).toMatchSnapshot();
		expect( container.innerHTML ).toContain( 'validation issues marked kept' );
		expect( container.querySelector( '.is-error' ) ).not.toBeNull();
	} );

	it( 'renders message when there are AMP validation errors', () => {
		setupUseSelect( {
			hasValidationErrors: true,
		} );

		act( () => {
			render( <AMPValidationStatusNotification />, container );
		} );

		expect( container.innerHTML ).toMatchSnapshot();
		expect( container.innerHTML ).toContain( 'issues needs review' );
		expect( container.querySelector( '.is-error' ) ).toBeNull();
	} );

	it( 'renders message when there are no errors and post is new', () => {
		setupUseSelect( {
			isEditedPostNew: true,
		} );

		act( () => {
			render( <AMPValidationStatusNotification />, container );
		} );

		expect( container.innerHTML ).toMatchSnapshot();
		expect( container.innerHTML ).toContain( 'issues will be checked for when the post is saved' );
		expect( container.querySelector( '.is-error' ) ).toBeNull();
	} );
} );

