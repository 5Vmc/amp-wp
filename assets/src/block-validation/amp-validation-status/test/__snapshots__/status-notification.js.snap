// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMPValidationStatusNotification renders error message when AMP compatibility is broken 1`] = `"<div class=\\"sidebar-notification is-error\\"><div class=\\"sidebar-notification__icon\\"><svg width=\\"21\\" height=\\"21\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g clip-path=\\"url(#clip-amp-validation-errors-kept)\\" fill=\\"#BB522E\\"><path d=\\"M10.762 2.541c4.4 0 8 3.6 8 8 0 1.6-.5 3-1.2 4.2l1.4 1.5c1.1-1.6 1.8-3.6 1.8-5.7 0-5.5-4.5-10-10-10-2 0-3.9.6-5.5 1.7l1.4 1.5c1.2-.8 2.6-1.2 4.1-1.2zM.762 10.541c0 5.5 4.5 10 10 10 2.7 0 5.1-1.1 6.9-2.8l-14-14.2c-1.8 1.8-2.9 4.3-2.9 7zm10 8c-4.4 0-8-3.6-8-8 0-1.5.4-2.8 1.1-4l10.9 10.9c-1.2.7-2.5 1.1-4 1.1z\\"></path><path d=\\"M14.262 9.74c.1 0 .1-.1.1-.2 0-.2-.2-.4-.4-.4h-2l1.6 1.6.7-1zM12.461 4.541h-.8l-1.6 2.6 1.7 1.7.7-4.3zM7.462 11.541s-.1.1-.1.2c0 .2.2.4.4.4h2.3l-.8 4.5h.7l2.6-4.1-3.5-3.6-1.6 2.6z\\"></path></g><defs><clipPath id=\\"clip-amp-validation-errors-kept\\"><path fill=\\"#fff\\" transform=\\"translate(.762 .541)\\" d=\\"M0 0h20v20H0z\\"></path></clipPath></defs></svg></div><div class=\\"sidebar-notification__content\\"><p>AMP blocked from validation issues marked kept.</p></div></div>"`;

exports[`AMPValidationStatusNotification renders message when there are AMP validation errors 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><div class=\\"amp-status-icon amp-status-icon--broken\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path fill=\\"#0075C2\\" d=\\"M13.3 9.1l-4 6.6h-.8l.7-4.3H7c-.2 0-.4-.2-.4-.4 0-.1.1-.2.1-.2l4-6.6h.7l-.7 4.3h2.2c.2 0 .4.2.4.4.1.1 0 .2 0 .2zM10 .5C4.7.5.4 4.8.4 10c0 5.3 4.3 9.5 9.6 9.5s9.6-4.3 9.6-9.5c0-5.3-4.3-9.5-9.6-9.5z\\"></path></svg></div></div><div class=\\"sidebar-notification__content\\"><p>AMP is working, but issues needs review.</p></div></div>"`;

exports[`AMPValidationStatusNotification renders message when there are no AMP validation errors 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><div class=\\"amp-status-icon \\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path fill=\\"#0075C2\\" d=\\"M13.3 9.1l-4 6.6h-.8l.7-4.3H7c-.2 0-.4-.2-.4-.4 0-.1.1-.2.1-.2l4-6.6h.7l-.7 4.3h2.2c.2 0 .4.2.4.4.1.1 0 .2 0 .2zM10 .5C4.7.5.4 4.8.4 10c0 5.3 4.3 9.5 9.6 9.5s9.6-4.3 9.6-9.5c0-5.3-4.3-9.5-9.6-9.5z\\"></path></svg></div></div><div class=\\"sidebar-notification__content\\"><p>AMP is working. All issues reviewed or removed.</p></div></div>"`;

exports[`AMPValidationStatusNotification renders message when there are no errors and post is new 1`] = `"<div class=\\"sidebar-notification\\"><div class=\\"sidebar-notification__icon\\"><div class=\\"amp-status-icon \\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\"><path fill=\\"#0075C2\\" d=\\"M13.3 9.1l-4 6.6h-.8l.7-4.3H7c-.2 0-.4-.2-.4-.4 0-.1.1-.2.1-.2l4-6.6h.7l-.7 4.3h2.2c.2 0 .4.2.4.4.1.1 0 .2 0 .2zM10 .5C4.7.5.4 4.8.4 10c0 5.3 4.3 9.5 9.6 9.5s9.6-4.3 9.6-9.5c0-5.3-4.3-9.5-9.6-9.5z\\"></path></svg></div></div><div class=\\"sidebar-notification__content\\"><p>Validation issues will be checked for when the post is saved.</p></div></div>"`;
